Mixpanel/MPLogger.h

```diff
@@ -7,34 +7,33 @@
 //
 
 #import <Foundation/Foundation.h>
+#import <asl.h>
 
-#ifndef MPLogger_h
-#define MPLogger_h
+static bool gLoggingEnabled = NO;
+static inline void MPSetLoggingEnabled(BOOL enabled) {
 
-static inline void MPLog(NSString *format, ...) {
-    __block va_list arg_list;
-    va_start (arg_list, format);
-    NSString *formattedString = [[NSString alloc] initWithFormat:format arguments:arg_list];
-    va_end(arg_list);
-    NSLog(@"[Mixpanel] %@", formattedString);
 }
 
-#ifdef MIXPANEL_ERROR
-#define MixpanelError(...) MPLog(__VA_ARGS__)
-#else
-#define MixpanelError(...)
-#endif
-
-#ifdef MIXPANEL_DEBUG
-#define MixpanelDebug(...) MPLog(__VA_ARGS__)
-#else
-#define MixpanelDebug(...)
-#endif
-
-#ifdef MIXPANEL_MESSAGING_DEBUG
-#define MessagingDebug(...) MPLog(__VA_ARGS__)
-#else
-#define MessagingDebug(...)
-#endif
-
-#endif
+#define __MP_MAKE_LOG_FUNCTION(LEVEL, NAME) \
+static inline void NAME(NSString *format, ...) { \
```

Comment:
is it possible that instead of having different methods for each level, to have the level be sent as a parameter and then use MPLog(Level.Error, "this is an error"). I think it would be cleaner that way if it's only one function for all logging